src/main/kotlin/com/terraformation/backend/customer/api/UsersController.kt

```diff
@@ -64,6 +62,19 @@ class UsersController(private val clock: InstantSource, private val userStore: U
           )
       userStore.updateUser(model)
       return SimpleSuccessResponsePayload()
+    } else if (user is FunderUser) {
+      val model =
+          user.copy(
+              countryCode = payload.countryCode,
+              emailNotificationsEnabled =
+                  payload.emailNotificationsEnabled ?: user.emailNotificationsEnabled,
+              firstName = payload.firstName,
+              lastName = payload.lastName,
+              locale = payload.locale?.let { Locale.forLanguageTag(it) },
+              timeZone = payload.timeZone,
+          )
```

Comment:
I assume this duplicate block (and the cookies-consented one) is what the "not sure if there's a better way" comment in the PR description was referring to.

I think this is fine as-is, but if you wanted to reduce duplication here, one way to do it would be to make your own "copy this object" method rather than using the auto-generated one (you'll need to call it something other than `copy`) and declare it in `TerrawareUser`.

Something like
```kotlin
interface Foo {
  val field1: String
  val field2: String
  fun duplicate(field1: String = this.field1, field2: String = this.field2): Foo
}

data class Bar(override val field1: String, override val field2: String) : Foo {
  override fun duplicate(field1: String, field2: String): Bar = Bar(field1, field2)
}
```

Reply:
That makes sense, and that's what I had seen from AI as well; just wanted to confirm in case there was some other way. I'll leave it for now and potentially add that in a future PR since I'll likely be needing to do the same more in the future.